data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -655,6 +660,49 @@ public void shutdown() {
     }
   }
 
+  /**
+   * flushRemainingBulkRequests Flush the remaining bulk requests during shutdown
+   * This is necessary to ensure that any remaining operations are sent to OpenSearch before the sink is shut down.
+   * It iterates through the bulkRequestMap, flushing each bulk request that has operations left.
+   * If an error occurs during the flush, it retries until sinkShutdownTimeout
+   * */
+  private void flushRemainingBulkRequests() {
+    if (bulkRequestMap.isEmpty()) {
+      return;
+    }
+
+    LOG.info("Flushing remaining {} bulk requests during shutdown", bulkRequestMap.size());
+    for (final Long threadId : bulkRequestMap.keySet()) {
+      final AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> bulkRequest = bulkRequestMap.get(threadId);
+      if (bulkRequest != null && bulkRequest.getOperationsCount() > 0) {
+        LOG.info("Flushing bulk request with {} operations for thread {}", bulkRequest.getOperationsCount(), threadId);
+		long retryTime = 0;
+        boolean success = false;
+        while (!success && retryTime <= FLUSH_TIMEOUT_WAIT_MS) {
+          try {
+            flushBatch(bulkRequest);
+            success = true;
+            LOG.info("Successfully flushed bulk request for thread {}", threadId);
+          } catch (Exception e) {
+            LOG.warn("Error flushing bulk request during shutdown retrying in 1 Seconds, remaining time: {}sec", (FLUSH_TIMEOUT_WAIT_MS - retryTime)/1000, e);
+            try {
+              	Thread.sleep( 1000L);
```

Comment:
This blocks flushing other bulk requests. Can this exception occur due to the content of the bulk requests? This would lead to one faulty bulk request blocking the others. If the exception can only occur for general reasons affecting all requests (e.g. OpenSearch down) this is fine.